tests/platform_helper/domain/test_codebase.py

```diff
@@ -0,0 +1,644 @@
+import filecmp
+import json
+import os
+from datetime import datetime
+from pathlib import Path
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+from unittest.mock import PropertyMock
+from unittest.mock import call
+from unittest.mock import patch
+
+import boto3
+import pytest
+import requests
+
+from dbt_platform_helper.domain.codebase import Codebase
+from dbt_platform_helper.domain.codebase import NotInCodeBaseRepositoryError
+from dbt_platform_helper.exceptions import ApplicationDeploymentNotTriggered
+from dbt_platform_helper.exceptions import ApplicationEnvironmentNotFoundError
+from dbt_platform_helper.exceptions import ApplicationNotFoundError
+from dbt_platform_helper.exceptions import CopilotCodebaseNotFoundError
+from dbt_platform_helper.exceptions import ImageNotFoundError
+from dbt_platform_helper.exceptions import NoCopilotCodebasesFoundError
+from dbt_platform_helper.utils.application import Environment
+from dbt_platform_helper.utils.git import CommitNotFoundError
+from tests.platform_helper.conftest import EXPECTED_FILES_DIR
+
+ecr_exceptions = boto3.client("ecr").exceptions
+ssm_exceptions = boto3.client("ssm").exceptions
+
+
+def mock_aws_client(get_aws_session_or_abort):
+    session = MagicMock(name="session-mock")
+    client = MagicMock(name="client-mock")
+    session.client.return_value = client
+    get_aws_session_or_abort.return_value = session
+    return client
+
+
+class CodebaseMocks:
+    def __init__(self, **kwargs):
+        self.load_application_fn = kwargs.get("load_application_fn", Mock())
+        self.get_aws_session_or_abort_fn = kwargs.get("get_aws_session_or_abort_fn", Mock())
+        self.input_fn = kwargs.get("input_fn", Mock(return_value="yes"))
+        self.echo_fn = kwargs.get("echo_fn", Mock())
+        self.confirm_fn = kwargs.get("confirm_fn", Mock(return_value=True))
+        self.check_codebase_exists_fn = kwargs.get(
+            "check_codebase_exists_fn",
+            Mock(
+                return_value="""
+                                             {
+                                                "name": "test-app", 
+                                                "repository": "uktrade/test-app",
+                                                "services": "1234"
+                                             }
+                                        """
+            ),
```

Comment:
Is there a reason for the whacky indenting above?

Reply:
black